crypto/x509/v3_addr.c

```diff
@@ -590,7 +590,12 @@ int X509v3_addr_add_prefix(IPAddrBlocks *addr,
 {
     IPAddressOrRanges *aors = make_prefix_or_range(addr, afi, safi);
     IPAddressOrRange *aor;
-    if (aors == NULL || !make_addressPrefix(&aor, a, prefixlen))
+    int maxprefixlen = length_from_afi(afi) * 8;
```

Comment:
I would have put this check into `make_addressPrefix()`.

The API is a bit undecided on whether it wants to support a generic AFI or only IPv4 and IPv6. In LibreSSL I have opted to cap the prefixlen to `8 * 16` unless the AFI is IPv4 in which case I have capped it at `8 * 4`. Your choice is legitimate, but a bit more restrictive.

Reply:
Yeah, ok - I've moved the check there.